# Statistical Validation of Churn Drivers

Exploratory Data Analysis identified several potential drivers of customer churn.  
To confirm whether these relationships are statistically significant, formal hypothesis testing was performed.

The following tests were conducted:

1. **Contract Type vs Churn — Chi-Square Test of Independence**
2. **Monthly Charges vs Churn — Independent Two-Sample T-Test**
3. **Tenure vs Churn — Independent Two-Sample T-Test**

These tests help determine whether the patterns observed during EDA represent **true statistical relationships rather than random variation**.

In [2]:
import pandas as pd

In [4]:
df = pd.read_csv("../data/telco_churn_cleaned_v1.csv")

In [5]:
df.head().T

,0,1,2,3,4
gender,Female,Male,Male,Male,Female
senior_citizen,0,0,0,0,0
partner,Yes,No,No,No,No
dependents,No,No,No,No,No
tenure,1,34,2,45,2
phone_service,No,Yes,Yes,No,Yes
multiple_lines,No phone service,No,No,No phone service,No
internet_service,DSL,DSL,DSL,DSL,Fiber optic
online_security,No,Yes,Yes,Yes,No
online_backup,Yes,No,Yes,No,No


## Hypothesis 1 — Contract Type and Churn

### Business Question
Does **contract type significantly influence customer churn?**

### Statistical Test
**Chi-Square Test of Independence**

This test determines whether there is a statistically significant relationship between **contract type** and **customer churn status**.

In [8]:
from scipy.stats import chi2_contingency

contingency_table = pd.crosstab(df['contract'], df['churn_flag'])

chi2, p, dof, expected = chi2_contingency(contingency_table)

print("Chi-square statistic:", chi2)
print("p-value:", p)

Chi-square statistic: 1184.5965720837926
p-value: 5.863038300673393e-258


### Results

| Metric | Value |
|------|------|
| Chi-Square Statistic | **1184.59** |
| P-Value | **5.86 × 10⁻²⁵⁸** |

### Interpretation

The extremely small p-value (p < 0.001) indicates a **statistically significant relationship** between contract type and customer churn.

Customers on **month-to-month contracts churn significantly more often** than customers with longer-term contracts.

### Business Implication

Contract structure is a major driver of churn risk.

Encouraging customers to transition from **month-to-month contracts to longer-term agreements** could significantly reduce churn and improve revenue stability.

## Hypothesis 2 — Monthly Charges and Churn

### Business Question
Do churned customers pay significantly different **monthly charges** compared to retained customers?

### Statistical Test
**Independent Two-Sample T-Test**

This test compares the average monthly charges between churned customers and retained customers.

In [9]:
from scipy.stats import ttest_ind

churned = df[df['churn_flag'] == 1]['monthly_charges']
retained = df[df['churn_flag'] == 0]['monthly_charges']

t_stat, p = ttest_ind(churned, retained)

print("T-statistic:", t_stat)
print("p-value:", p)

T-statistic: 16.536738015936308
p-value: 2.706645606888385e-60


### Results

| Metric | Value |
|------|------|
| T-Statistic | **16.54** |
| P-Value | **2.71 × 10⁻⁶⁰** |

### Interpretation

The p-value is far below 0.001, indicating a **statistically significant difference** in monthly charges between churned and retained customers.

Customers who churn tend to have **higher monthly charges**.

### Business Implication

Higher pricing may contribute to churn due to:

- Price sensitivity
- Lower perceived value
- Availability of competitive alternatives

Pricing strategies and perceived value delivery should be evaluated to reduce churn risk.

## Hypothesis 3 — Tenure and Churn

### Business Question
Does **customer tenure influence the likelihood of churn?**

### Statistical Test
**Independent Two-Sample T-Test**

This test compares the tenure of churned customers and retained customers.

In [10]:
churned_tenure = df[df['churn_flag'] == 1]['tenure']
retained_tenure = df[df['churn_flag'] == 0]['tenure']

t_stat, p = ttest_ind(churned_tenure, retained_tenure)

print("T-statistic:", t_stat)
print("p-value:", p)

T-statistic: -31.57955051135377
p-value: 7.999057960590132e-205


### Results

| Metric | Value |
|------|------|
| T-Statistic | **-31.58** |
| P-Value | **7.99 × 10⁻²⁰⁵** |

### Interpretation

The extremely low p-value indicates a **highly significant difference in tenure** between churned and retained customers.

Customers who churn tend to have **significantly lower tenure**.

### Business Implication

Churn risk is highest during the **early stages of the customer lifecycle**.

Retention strategies should prioritize:

- Customers within their **first 12 months**
- Early engagement initiatives
- Proactive customer support and onboarding

## Key Takeaways

The statistical validation confirms that multiple factors significantly influence customer churn.

Key insights include:

- Customers on **month-to-month contracts** show significantly higher churn probability.
- Customers paying **higher monthly charges** are more likely to leave.
- Customers with **shorter tenure** exhibit the highest churn risk.

These findings validate the patterns observed during exploratory analysis and reinforce the importance of focusing retention efforts on **high-risk customer segments**.

## Summary of Statistical Findings

| Factor | Statistical Test | Result | Business Insight |
|------|------|------|------|
| Contract Type | Chi-Square | Significant | Month-to-month customers exhibit the highest churn rates |
| Monthly Charges | Independent T-Test | Significant | Higher monthly charges increase churn risk |
| Tenure | Independent T-Test | Significant | Customers in early lifecycle stages are more likely to churn |

These statistical tests confirm that the churn patterns observed during exploratory analysis are **statistically significant**.

The results highlight three primary drivers of churn:

- Contract structure  
- Pricing levels  
- Customer tenure  

These findings support targeted retention strategies focused on **early customer lifecycle engagement and contract optimization**.